test/unit/solver/test_cuda_rosenbrock.cpp

```diff
@@ -0,0 +1,205 @@
+                         .reactants({ bar })
+                         .products({ yields(foo, 1), yields(quz, 1.4) })
+                         .phase(gas_phase)
+                         .rate_constant(micm::ArrheniusRateConstant({ .A_ = 1.0e-6 }));
+
+  micm::Process r3 = micm::Process::create().reactants({ quz }).products({}).phase(gas_phase).rate_constant(
+      micm::ArrheniusRateConstant({ .A_ = 3.5e-6 }));
+
+  return micm::CudaRosenbrockSolver<MatrixPolicy, SparseMatrixPolicy, LinearSolverPolicy>(
+      micm::System(micm::SystemParameters{ .gas_phase_ = gas_phase }),
+      std::vector<micm::Process>{ r1, r2, r3 },
+      micm::RosenbrockSolverParameters::three_stage_rosenbrock_parameters(number_of_grid_cells, false));
+}
+
+template<template<class> class MatrixPolicy, template<class> class SparseMatrixPolicy, class LinearSolverPolicy>
+micm::RosenbrockSolver<MatrixPolicy, SparseMatrixPolicy, LinearSolverPolicy> getSolver_cpu(std::size_t number_of_grid_cells)
+{
+  // ---- foo  bar  baz  quz  quuz
+  // foo   0    1    2    -    -
+  // bar   3    4    5    -    -
+  // baz   6    -    7    -    -
+  // quz   -    8    -    9    -
+  // quuz 10    -   11    -    12
+
+  auto foo = micm::Species("foo");
+  auto bar = micm::Species("bar");
+  auto baz = micm::Species("baz");
+  auto quz = micm::Species("quz");
+  auto quuz = micm::Species("quuz");
+
+  micm::Phase gas_phase{ std::vector<micm::Species>{ foo, bar, baz, quz, quuz } };
+
+  micm::Process r1 = micm::Process::create()
+                         .reactants({ foo, baz })
+                         .products({ yields(bar, 1), yields(quuz, 2.4) })
+                         .phase(gas_phase)
+                         .rate_constant(micm::ArrheniusRateConstant({ .A_ = 2.0e-11, .B_ = 0, .C_ = 110 }));
+
+  micm::Process r2 = micm::Process::create()
+                         .reactants({ bar })
+                         .products({ yields(foo, 1), yields(quz, 1.4) })
+                         .phase(gas_phase)
+                         .rate_constant(micm::ArrheniusRateConstant({ .A_ = 1.0e-6 }));
+
+  micm::Process r3 = micm::Process::create().reactants({ quz }).products({}).phase(gas_phase).rate_constant(
+      micm::ArrheniusRateConstant({ .A_ = 3.5e-6 }));
+
+  return micm::RosenbrockSolver<MatrixPolicy, SparseMatrixPolicy, LinearSolverPolicy>(
+      micm::System(micm::SystemParameters{ .gas_phase_ = gas_phase }),
+      std::vector<micm::Process>{ r1, r2, r3 },
+      micm::RosenbrockSolverParameters::three_stage_rosenbrock_parameters(number_of_grid_cells, false));
+}
+
+template<template<class> class MatrixPolicy, template<class> class SparseMatrixPolicy, class LinearSolverPolicy>
+std::vector<double> get_jacobian_cpu(size_t number_of_grid_cells){
+  auto cpu_solver = getSolver_cpu<MatrixPolicy, SparseMatrixPolicy, LinearSolverPolicy>(number_of_grid_cells); 
+  auto jacobian = cpu_solver.GetState().jacobian_;
+  EXPECT_GE(jacobian.AsVector().size(), 13 * number_of_grid_cells);
```

Comment:
Could you please briefly explain why we expect `EXPECT_GE` here?

Reply:
This is part of the testing on the CPU code. I just wanted to make sure it follow the same testing.

Reply:
Cool. Then maybe @mattldawson or @K20shores can explain more here?

Reply:
I have zero clue here.

Reply:
If the vectorizable dimension of the matrix classes is not a multiple of the number of grid cells, you may end up with an underlying vector that is larger than you need. For the JIT and CUDA classes, we force the vectorizable dimension to equal the number of grid cells, and we can make this requirement for all the classes in the future if we want to.

Reply:
Thanks @mattldawson for your clarification. That makes sense to me now.